view/frontend/web/template/payment/pos-cloud-form.html

```diff
@@ -50,38 +50,69 @@
+                                            options: getConnectedTerminals(),
+                                            optionsValue: 'value',
+                                            optionsText: 'key',
+                                            value: terminalId"
+                            >
+                    </select>
+                </div>
             </div>
-        </div>
 
-        <div class="checkout-agreements-block">
-            <!-- ko foreach: $parent.getRegion('before-place-order') -->
-            <!-- ko template: getTemplate() --><!-- /ko -->
-            <!--/ko-->
-        </div>
+            <!-- ko if: (hasInstallments())-->
+            <div class="field required"
+                 data-bind="attr: {id: getCode() + '_installments_div'}, visible: installments().length > 0">
+                <label data-bind="attr: {for: getCode() + '_installments'}" class="label">
+                    <span><!-- ko text: $t('Installments')--><!-- /ko --></span>
+                </label>
+                <div class="control">
+
+                    <select class="select"
+                            name="payment[number_of_installments]"
+                            data-bind="attr: {id: getCode() + '_installments', 'data-container': getCode() + '-installments', 'data-validate': JSON.stringify({required:false})},
+
+                                            enable: isActive($parents),
+                                            options: installments,
+                                            optionsValue: 'value',
+                                            optionsText: 'key',
+                                            optionsCaption: $t('Do not use Installments'),
+                                            value: installment"
+                            data-validate="{required:true}">
+                    </select>
+                </div>
+            </div>
+            <!-- /ko -->
+
+            <div class="checkout-agreements-block">
+                <!-- ko foreach: $parent.getRegion('before-place-order') -->
+                <!-- ko template: getTemplate() --><!-- /ko -->
+                <!--/ko-->
+            </div>
+        <!--/ko-->
+        <!-- ko ifnot: (getConnectedTerminals().length > 0) -->
+            <p><!-- ko text: $t('Please connect at least 1 terminal')--><!-- /ko --></p>
+        <!--/ko-->
+
         <div class="actions-toolbar">
             <div class="primary">
                 <button class="action primary checkout"
                         type="submit"
                         data-bind="
                             click: initiate,
                             attr: {title: $t('Place Order')},
-                            enable: getConnectedTerminals().length > 0
+                            enable: isButtonActive()
```

Comment:
should we not check if there is any connected terminals to select before making the button active ?
What will happen if there are no connected terminals available ?

Reply:
indeed, I will put this check in the isButtonActive function